src/javascript/typescript-config.ts

```diff
@@ -477,6 +547,24 @@ export class TypescriptConfig extends Component {
     if (project instanceof NodeProject) {
       project.npmignore?.exclude(`/${fileName}`);
     }
+
+    if (options.extends) {
+      const values = Array.isArray(options.extends)
+        ? options.extends
+        : [options.extends];
+      values.map((value) => this.addExtends(value));
+    }
+  }
+
+  public addExtends(value: string | TypescriptConfig) {
```

Comment:
```suggestion
  public addExtends(value: string | TypescriptConfig) {
```
Same as above re union type.